src/core/base_stream.js

```diff
@@ -44,6 +44,22 @@ class BaseStream {
     unreachable("Abstract method `getBytes` called");
   }
 
+  async getImageData(length, ignoreColorSpace) {
```

Comment:
Can we please add a comment about when this method is safe to use, since that might not otherwise be apparent to a future reader?
```suggestion
  /**
   * NOTE: This method can only be used to get image-data that is guaranteed
   *       to be fully loaded, since otherwise intermittent errors may occur;
   *       note the `ObjectLoader` class.
   */
  async getImageData(length, ignoreColorSpace) {
```